sn_networking/src/event.rs

```diff
@@ -745,18 +797,42 @@ impl SwarmDriver {
                         .send(Err(Error::RecordNotFound))
                         .map_err(|_| Error::InternalMsgChannelDropped)?;
                 }
+
+                // return the accumulated cache candidates, if any
+                if let Some((record_key, stored_cache_candidates)) =
+                    self.pending_get_record_for_cache_candidates.remove(&id)
+                {
+                    if !stored_cache_candidates.is_empty() {
+                        // log the pending_get_record* stats; get_record_for_cache should not explode and should always be
+                        // greater or equal to get_record
+                        debug!(
+                            "pending_get_record has got {} entries",
+                            self.pending_get_record.len()
+                        );
+                        debug!(
+                            "pending_get_record_for_cache_candidates has got {} entries",
+                            self.pending_get_record_for_cache_candidates.len()
```

Comment:
why do we need `pending_get_record_for_cache_candidates` ? can we not just directly fire the event there?

Reply:
The `RecordKey` is not present when the `KademliaEvent` are emitted (except for when record is found). We need the RecordKey to inform the cache_candidates.

Expanding the `pending_get_record` to store the `RecordKey` is complicated, as the flow terminates at a lot of places. An extra field was the simplest approach.

Reply:
Also there is a `kademlia.query(id)` function that we can use to get the `RecordKey`, but the query is cleaned up by kad when it completes. Thus, cannot use that one aswell